run.sh

```diff
@@ -11,13 +11,33 @@
 readonly BINARY_NAME="${__RALLY_INTERNAL_BINARY_NAME}"
 readonly HUMAN_NAME="${__RALLY_INTERNAL_HUMAN_NAME}"
 
+function install_esrally_with_setuptools() {
+    # Check if optional parameter with Rally binary path, points to an existing executable file.
+    if [[ -n $1 ]]; then
+     if [[ -f $1 && -x $1 ]]; then return; fi
+    fi
+
+    if [[ ${IN_VIRTUALENV} == 0 ]]; then
+        python3 setup.py -q develop --user
+    else
+        python3 setup.py -q develop
+    fi
+}
+
 # Attempt to update Rally itself by default but allow user to skip it.
 SELF_UPDATE=YES
 # Assume that the "main remote" is called "origin"
 REMOTE="origin"
 
 # While we could also check via the presence of `VIRTUAL_ENV` this is a bit more reliable.
-python3 -c 'import sys; print(sys.real_prefix)' >/dev/null 2>&1 && IN_VIRTUALENV=1 || IN_VIRTUALENV=0
+# Check for both pyvenv and normal venv environments
+# https://legacy.python.org/dev/peps/pep-0405/
+if python3 -c 'import sys; sys.exit(0) if ".pyenv" in sys.base_prefix else exit(0) if hasattr(sys,"real_path") else exit(1)' >/dev/null 2>&1
```

Comment:
I think the following would be a bit simpler? 

```python
import sys; exit(0) if ".pyenv" in sys.base_prefix or hasattr(sys,"real_path") else exit(1)
```

Reply:
Changing the logic entirely here as per our discussion.